frontend/src/components/TransactionForm/index.js

```diff
@@ -39,6 +44,11 @@ const TransactionForm = ({item, setItemsToEdit}) => {
 
   const onSubmit = async (evt) => {
     evt.preventDefault();
+
+    if (!user) {
```

Comment:
visitors or unlogged users won't have access to such component anyways, right?
there is no need for this check I reckon

another point in this filem in the initiation of states in the first lines, consider using the optional chaining operator
(item && item.title) || "" ---becomes--> item?.title || ''